crates/ra_hir/src/ty/method_resolution.rs

```diff
@@ -166,7 +166,19 @@ pub(crate) fn lookup_method(
     name: &Name,
     resolver: &Resolver,
 ) -> Option<(Ty, Function)> {
-    iterate_method_candidates(ty, db, resolver, Some(name), |ty, f| Some((ty.clone(), f)))
+    iterate_method_candidates(ty, db, resolver, Some(name), LookupMode::MethodCall, |ty, f| {
+        if let AssocItem::Function(f) = f {
+            Some((ty.clone(), f))
+        } else {
+            None
+        }
+    })
+}
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq)]
+pub enum LookupMode {
```

Comment:
Is this what we're looking up? A comment would be good.

Reply:
Yeah, done